scipy/stats/tests/test_morestats.py

```diff
@@ -277,6 +277,16 @@ def test_gumbel_r(self):
         assert_array_less(A1, crit1[-2:])
         assert_(A2 > crit2[-1])
 
+    def test_weibull_min(self):
+        x1 = np.array([225, 171, 198, 189, 135, 162, 135, 117, 162])
```

Comment:
where does this testcase come from?

Reply:
both tests are from article where this method was introduced

Reply:
please add a reference, too. and can the test be improved to check that A1 is approximately 0.26, c = 0.42 ...

Reply:
With current quality of estimation by MLE we can't estimate that c = 0.42

Reply:
hm, that does not sound very comforting. do we get at least close to A1 = 0.26 and c = 0.42? what is the output of the function?

Reply:
example 1:
paper: loc = 99.02, scale = 78.23, c = 2.38, paper_c = 0.42, A2 = 0.26 < 0.301
scipy: loc = -0.10, scale = 181.5, c = 5.98, paper_c = 0.167, A2 = 0.223 < 0.301

example 2:
paper: loc = 9.313, scale= 93.50, c = 0.763, paper_c = 1.31, A2 = 0.54 > 0.342
scipy: loc = 11.99, scale = 100.7, c = 0.75, paper_c = 1.33, A2 = 2.29 > 0.342

In the paper they introduced a bias correction for 3-param weibull estimation, as a separate feature I could code the specific .fit function for weibull_min

Reply:
thanks for the clarification. after checking the paper, I understand that fitting the parameters is not easy and the MLE estimate is known to have problems. that is why the authors introduce a method for three cases A, B and C. so I assume the approach implemented might not be very robust. what do you think about this point?

Reply:
gh-11806 is an example of the trouble.

> In the paper they introduced a bias correction for 3-param weibull MLE, as a separate feature I could code the specific .fit function for weibull_min

That would be really nice.

Reply:
Fitting is not foolproof, but it's reasonably reliable for Weibull-distributed data with moderate values of the parameters. 

![image](https://user-images.githubusercontent.com/6570539/187015007-ca7f7a54-8465-49ab-8186-171dafdc3bdf.png)

I think that we only really need to distinguish between case A and case B/C in this PR. As I'll explain, we only really need to concern ourselves with case A.

For case C, the "likelihood steadily decreases as [the location] moves away from [the minimum of the data] towards $-\infty$", so the maximum likelihood occurs at the maximum possible value, which is constrained to be the minimum of the data. The authors also explicitly state that the minimum value of the data is the MLE of location in case B:
![image](https://user-images.githubusercontent.com/6570539/187015223-e740a0f2-cb3a-4c8b-8dac-143055feb6cb.png)
So case A is all cases in which the MLE of the location is not at the minimum of the data. 

I don't think the authors make it very clear, but case A tends to correspond with values of the shape parameter $m > 1$, and B/C is for $m < 1$. This makes sense from the plot of the PDF :
![image](https://user-images.githubusercontent.com/6570539/187015386-c3492e44-82e0-4905-abd6-1a0b00f1f54e.png)

For small shape parameter, the PDF is highest toward the left end of the support, which is why it makes sense for `loc` to be at the minimum of the data.

This is important because the authors did not actually tabulate critical values for cases in which the shape parameter is less than 2, and I've observed that - even with their bias reduction procedure - the authors' recommendation to use the critical values for $m=2$ when $m < 2$ results in a very unconservative test. So I think whenever we find that the shape parameter $m < 2$, we should just report to the user that critical values are not available because they are below the tabulated range. We only need to provide a fit for and complete the test for case A; otherwise, we error out or report NaNs or something.